docs/Developers/NextGen_Core.rst

```diff
@@ -310,12 +310,23 @@ in case of code clone detection).
 Superior Caching
 ----------------
 
-The NextGen-Core's caching mechanism is based on task objects. Bears can offload
-tasks via `generate_tasks()` which get executed by a Python pool. Structure wise
-the cache is a dictionary-like-object with bear types and cache-tables as key
-value pairs. The cache-tables themselves are dictionary-like-objects that map
-the hash values of the task objects (generated by
-``PersistentHash.persistent_hash``) to the bear results.
+NextGen-Core's caching can be broken up into two parts:
+
+- Memoization of ``FileFactory`` objects.
+- Caching of task objects.
+
+Since NextGen-Core provides coala to interface with the files using the
+``FileFactory``, ``FileFactory`` uses its own caching mechanism to ensure
+high performance I/O operations. The file contents are accessed using
+properties which are memoized so that whenever these properties are accessed
+instead of loading the files again the cached results are returned.
```

Comment:
a bit reordering makes it easier to read:
```
so that whenever these properties are accessed the cached results are returned instead of loading the files again.
```